### PR TITLE
release: v1.23.3

### LINKED ISSUE #11048: [WIP]Lotus 1.23.3
# Lotus 1.23.3 Release

## 🚢 Estimated shipping date

The estimated shipping dates for the Lotus v1.23.3 release are as follows:

2023-07-04: v1.23..3-rc1
2023-08-10: v1.23.3

## 🔦 Highlights

< See Changelog>

## ✅ Release Checklist

**Note for whomever is owning the release:** please capture notes as comments in this issue for anything you noticed that could be improved for future releases.  There is a *Post Release* step below for incorporating changes back into the [RELEASE_ISSUE_TEMPLATE](https://github.com/filecoin-project/lotus/blob/master/documentation/misc/RELEASE_ISSUE_TEMPLATE.md), and this is easier done by collecting notes from along the way rather than just thinking about it at the end.

First steps:

  - [x] Fork a new branch (`release/vX.Y.Z`) from `master` and make any further release related changes to this branch. If any "non-trivial" changes get added to the release, uncheck all the checkboxes and return to this stage.
  - [x] Bump the version in `build/version.go` in the `master` branch to `vX.Y.(Z+1)-dev` (bump from feature release) or `vX.(Y+1).0-dev` (bump from mandatory release). Run make gen and make docsgen-cli before committing changes
    
Prepping an RC:

- [x] version string in `build/version.go` has been updated (in the `release/vX.Y.Z` branch)
- [x] run `make gen && make docsgen-cli`
- [x] Generate changelog using the script at scripts/mkreleaselog
- [x] Add contents of generated text to lotus/CHANGELOG.md in addition to other details
- [ ] tag commit with `vX.Y.Z-rcN`
- [ ] cut a pre-release [here](https://github.com/filecoin-project/lotus/releases/new?prerelease=true)

Testing an RC:

- [x] **Stage 0 - Automated Testing**
  - Automated Testing
    - [ ] CI: Ensure that all tests are passing.
    - [ ] Testground tests

- [x] **Stage 1 - Internal Testing**
  - Binaries
    - [ ] Ensure the RC release has downloadable binaries
  - Upgrade our testnet infra
    - [ ] Wait 24 hours, confirm nodes stay in sync
  -  Upgrade our mainnet infra
    - [ ] Subset of development full archival nodes
    - [ ] Subset of bootstrappers (1 per region)
    - [ ] Confirm nodes stay in sync
    - Metrics report
        - Block validation time
        - Memory / CPU usage
        - Number of goroutines
        - IPLD block read latency
        - Bandwidth usage
    - [ ] If anything has worsened significantly, investigate + fix
  - Confirm the following work (some combination of Testground / Calibnet / Mainnet / beta users)
    - [ ] Seal a sector
    - [ ] make a deal
    - [ ] Submit a PoSt
    - [ ] (optional) let a sector go faulty, and see it be recovered
    
- [ ] **Stage 2 - Community Testing**
  - [ ] Test with [SPX](https://github.com/filecoin-project/lotus/discussions/7461) fellows
  - [ ] Work on documentations for new features, configuration changes and so on.

- [ ] **Stage 3 - Community Prod Testing**
  - [ ] Update the [CHANGELOG.md](https://github.com/filecoin-project/lotus/blob/master/CHANGELOG.md) to the state that can be used as release note.
  - [ ] Invite the wider community through (link to the release issue)
    
- [ ] **Stage 4 - Stable Release**
  - [ ] Final preparation
    - [x] Verify that version string in [`version.go`](https://github.com/filecoin-project/lotus/blob/master/build/version.go) has been updated.
    - [x] Verify that codegen is up to date (`make gen && make docsgen-cli`)
    - [x] Ensure that [CHANGELOG.md](https://github.com/filecoin-project/lotus/blob/master/CHANGELOG.md) is up to date
    - [x] Merge `release-vX.Y.Z` into the `releases` branch.
    - [x] Tag this merge commit (on the `releases` branch) with `vX.Y.Z`
    - [x] Cut the release [here](https://github.com/filecoin-project/lotus/releases/new?prerelease=false&target=releases).

- [ ] **Post-Release**
  - [ ] Merge the `releases` branch back into `master`, ignoring the changes to `version.go` (keep the `-dev` version from master). Do NOT delete the `releases` branch when doing so!
  - [ ] Update [RELEASE_ISSUE_TEMPLATE.md](https://github.com/filecoin-project/lotus/blob/master/documentation/misc/RELEASE_ISSUE_TEMPLATE.md) with any improvements determined from this latest release iteration.
  - [ ] Create an issue using [RELEASE_ISSUE_TEMPLATE.md](https://github.com/filecoin-project/lotus/blob/master/documentation/misc/RELEASE_ISSUE_TEMPLATE.md) for the _next_ release.

## ❤️ Contributors

See the final release notes!

## ⁉️ Do you have questions?

Leave a comment in this ticket!

### PR DESCRIPTION
**Pull request by arajasek**
## Related Issues

Closes #11048

## Proposed Changes

Merge the release branch into releases, from where we'll be cutting the final release.

## Checklist

Before you mark the PR ready for review, please make sure that:

- [ ] Commits have a clear commit message.
- [ ] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

File: cmd/lotus/daemon.go [hunk 1]
```diff
@@ -574,3 +603,122 @@ func ImportChain(ctx context.Context, r repo.Repo, fname string, snapshot bool)
 
 	return nil
 }
+
+func slashConsensus(a lapi.FullNode, p string, from string) error {
```
Comment (magik6k) [inline cmd/lotus/daemon.go:607]: This really doesn't belong in this file. Should probably be a separate DI module, enabled with config, not a random flag on one of the main commands.

**Review by magik6k — APPROVED** [review]
The slasher thing should be moved around, rest looks good

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,107 @@
 # Lotus changelog
 
+# UNRELEASED
```
Comment (jennijuju) [inline CHANGELOG.md:3]: is this line still suppose to be here?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,107 @@
...
+
+# v1.23.3 / 2023-07-27
+
+This feature release of Lotus includes numerous improvements and enhancements for node operators, ETH RPC-providers and storage providers.
+
+## Highlights
```
Comment (jennijuju) [inline CHANGELOG.md:9]: missing highlight this ? https://www.notion.so/Write-explainer-doc-for-SPs-183a7775406d451a8b19e3cdc2cdf5c3?d=d8df4e142b78427ab470db68877da1ba&pvs=4#abba19208e3b4420a6388df7a3a48b62

Comment (jennijuju) [reply CHANGELOG.md:9]: come discussion link will on Monday

**Review by jennijuju — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,107 @@
...
+
+- The Ethereum-like RPC exposed by Lotus is now compatible with EIP-1898: https://github.com/filecoin-project/lotus/pull/10815
+- The lotus-miner PieceReader now supports parallel reads: https://github.com/filecoin-project/lotus/pull/10913
+- Added new environment variable `LOTUS_EXEC_TRACE_CACHE_SIZE` to configure execution trace cache size ([filecoin-project/lotus#10585](https://github.com/filecoin-project/lotus/pull/10585))
+  - If unset, we default to caching 16 most recent execution traces. Storage Providers may want to set this to 0, while exchanges may want to crank it up. 
+
+## New features
+  - Implement a tooling for slasher  ([filecoin-project/lotus#10928](https://github.com/filecoin-project/lotus/pull/10928))
```
Comment (jennijuju) [inline CHANGELOG.md:17]: I think this deserves a highlight(one of the big goal we are trying to hit) & link to the use educational context

File: CHANGELOG.md (hunk 2 above)
Comment (arajasek) [reply CHANGELOG.md:3]: Yes, it should always be there for any UNRELEASED changes

File: CHANGELOG.md (hunk 3 above)
Comment (arajasek) [reply CHANGELOG.md:9]: #11098 has been merged

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,114 @@
...
+# v1.23.3 / 2023-08-01
+
+This feature release of Lotus includes numerous improvements and enhancements for node operators, ETH RPC-providers and storage providers.
+
+## Highlights
+
+- [Lotus now includes a Slasher tool](https://github.com/filecoin-project/lotus/pull/10928) to monitor the network for Consensus Faults, and report them as appropriate
+  - The Slasher investigates all incoming blocks, and assesses whether they trigger any of the three Consensus Faults defined in the Filecoin protocol
+  - If any faults are detected, the Slasher sends a `ReportConsensusFault` message to the faulty miner
+  - For more information on the Slasher, including how to run it, please see LINK
```
Comment (arajasek) [inline CHANGELOG.md:14]: TODO: link after updating docs with changes in #11126

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,114 @@
...
+- The Ethereum-like RPC exposed by Lotus is now compatible with EIP-1898: https://github.com/filecoin-project/lotus/pull/10815
+- The lotus-miner PieceReader now supports parallel reads: https://github.com/filecoin-project/lotus/pull/10913
+- Added new environment variable `LOTUS_EXEC_TRACE_CACHE_SIZE` to configure execution trace cache size ([filecoin-project/lotus#10585](https://github.com/filecoin-project/lotus/pull/10585))
+  - If unset, we default to caching 16 most recent execution traces. Storage Providers may want to set this to 0, while exchanges may want to crank it up. 
+- The tracer's Elastic Search transport has been upgraded to support batch traces: [filecoin-project/lotus#10405](https://github.com/filecoin-project/lotus/pull/10405)
+  - This facilitates an experiment being run by ProbeLab [to measure the performance of the Gossipsub protocol in the Filecoin network](https://www.notion.so/pl-strflt/Write-explainer-doc-for-SPs-183a7775406d451a8b19e3cdc2cdf5c3?d=d8df4e142b78427ab470db68877da1ba#abba19208e3b4420a6388df7a3a48b62).
```
Comment (jennijuju) [inline CHANGELOG.md:20]: replace link with https://github.com/filecoin-project/lotus/discussions/11118

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,114 @@
...
+- The lotus-miner PieceReader now supports parallel reads: https://github.com/filecoin-project/lotus/pull/10913
+- Added new environment variable `LOTUS_EXEC_TRACE_CACHE_SIZE` to configure execution trace cache size ([filecoin-project/lotus#10585](https://github.com/filecoin-project/lotus/pull/10585))
+  - If unset, we default to caching 16 most recent execution traces. Storage Providers may want to set this to 0, while exchanges may want to crank it up. 
+- The tracer's Elastic Search transport has been upgraded to support batch traces: [filecoin-project/lotus#10405](https://github.com/filecoin-project/lotus/pull/10405)
+  - This facilitates an experiment being run by ProbeLab [to measure the performance of the Gossipsub protocol in the Filecoin network](https://www.notion.so/pl-strflt/Write-explainer-doc-for-SPs-183a7775406d451a8b19e3cdc2cdf5c3?d=d8df4e142b78427ab470db68877da1ba#abba19208e3b4420a6388df7a3a48b62). 
+  - Users are invited to opt into the experiment by following the instructions [here](https://www.notion.so/pl-strflt/Write-explainer-doc-for-SPs-183a7775406d451a8b19e3cdc2cdf5c3?pvs=4#092a3a68d6734a12961f5c3fe95767e5)
```
Comment (jennijuju) [inline CHANGELOG.md:21]: same as above - can we give a tldr on why we encourage user to run this please?

**Review by jennijuju — APPROVED** [review]

**Pull request merged**